lib/super_digit.rb

```diff
@@ -1,15 +1,45 @@
 # Superdigit
 
-# Time Complexity - ?
-# Space Complexity - ?
+def digit_helper(n)
+  sum = 0
+  until n.zero? do
+    sum += n % 10
+    n = n/10
+  end
+
+  return sum
+end
+
+def recursive_digit_helper(n)
+  return n if n < 10
+
+  return n % 10 + recursive_digit_helper(n / 10)
+end
+
+# Time Complexity - O(n)
+# Space Complexity - O(log(n))
+
 def super_digit(n)
```

Comment:
This works but is O(d^2) where d is the number of digits, because your helper is O(d) and it's done d times due to recursion.